Add script for installing dependencies

dpino opened the pull request:
Tested on Ubuntu.

Also fix an import with Python3 (I think it is expected now that the script run using Python3, right?).

psaavedra commented on install-dependencies.sh:
```diff
@@ -0,0 +1,53 @@
+)
+
+pip_modules=(
+    matrix_client
```
pip install relies in https://github.com/psaavedra/matrix-bot/blob/main/setup.py. It is important to keep the track of the installed versions like the setup.py does: https://github.com/psaavedra/matrix-bot/blob/main/setup.py#L59

I think you could do a similar thing with something like this:

```
pip_modules=(
    matrix-client==0.0.6
    ...
)
```

But still I dont like the idea of to have the dependencies listed in two files at the same time.

dpino replied:
I agree, it's better to keep the dependencies in one place. I was not aware of this `setup.py` file.

Mainly the problem I was trying to solve is that whenever I run the matrix-bot from a new host, I spend a few minutes solving dependencies until the bot runs successfully. So I would like to have a script that I can run and got all the dependecies solved.

So it occurs to me to keep this install-dependencies.sh script for required packages only (not Python modules), and then allow `setup.py` to be executed directly to install de Python modules. WDYT?

psaavedra replied:
Sounds reasonable.

dpino commented:
I updated the PR with the following changes:
  - `install-package-dependencies.sh` only installs package dependencies (not modules).
  - Moved manual installation steps stated in README.md to `setup.py`.
  - Updated README.md.

psaavedra commented on setup.py:
```diff
@@ -34,6 +36,11 @@ def exec_file(path_segments):
     pass
 
+class PostInstallCommand(install):
+    def run(self):
```
I dont like this idea because the goal of the install command of the setuptools is other. You can try to install the package. You can try to test the results of to understand  it better:

```
python setup.py develop
python setup.py install --user
```

psaavedra commented:
What do you thing about this?

``` diff
diff --git a/requirements.txt b/requirements.txt
new file mode 100644
index 0000000..b303b41
--- /dev/null
+++ b/requirements.txt
@@ -0,0 +1,8 @@
+getconf==1.5.1
+feedparser==6.0.2
+matrix-client==0.0.6
+python-dateutil==2.8.1
+python-ldap==3.1.0
+python-memcached==1.59
+pytz==2018.9
+requests==2.21.0
diff --git a/setup.py b/setup.py
index 5dce293..500c797 100755
--- a/setup.py
+++ b/setup.py
@@ -3,6 +3,10 @@ import os
 
 here = os.path.abspath(os.path.dirname(__file__))
 
+try: # for pip >= 10
+    from pip._internal.req import parse_requirements
+except ImportError: # for pip <= 9.0.3
+    from pip.req import parse_requirements
 
 def read_file(path_segments):
     """Read a file from the package. Takes a list of strings to join to
@@ -56,16 +60,7 @@ setup(
         "tools/matrix-subscriber",
     ],
     zip_safe=False,
-    install_requires=[
-        "getconf==1.5.1",
-        "matrix-client==0.4.0",
-        "python-ldap==3.1.0",
-        "python-memcached==1.59",
-        "feedparser==6.0.2",
-        "pytz==2018.9",
-        "requests==2.21.0",
-        "python-dateutil==2.8.1",
-    ],
+    install_requires = list(map(lambda x:x.requirement,parse_requirements('requirements.txt', session=''))),
 
     download_url='https://github.com/psaavedra/matrix-bot/zipball/master',
     classifiers=[

```


We could move the dependencies to a `requirements.txt`. Then, you can use python just `pip install -r requirements.txt` in the `install-package-dependencies.sh` script.

psaavedra commented:
Check this: https://github.com/psaavedra/matrix-bot/pull/62/commits/5ee2cb30d37447a9df5363a81ac740abb3562ece

psaavedra commented on install-dependencies.sh:
```diff
@@ -0,0 +1,43 @@
+}
+
+install_module_dependencies() {
+    pip3 install $(cat requirements.txt)
```
why not `python pip install -r requirements.txt`. it sounds much more canonical.
